api/v1alpha1/ratelimit_types.go

```diff
@@ -186,14 +190,27 @@ type RateLimitCostMetadata struct {
 type RateLimitSelectCondition struct {
 	// Headers is a list of request headers to match. Multiple header values are ANDed together,
 	// meaning, a request MUST match all the specified headers.
-	// At least one of headers or sourceCIDR condition must be specified.
+	// At least one of headers or method or path or sourceCIDR condition must be specified.
 	//
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	Headers []HeaderMatch `json:"headers,omitempty"`
 
+	// Method is the request method to match. If not specified, it matches all methods.
+	// At least one of headers or method or path or sourceCIDR condition must be specified.
+	//
+	// +optional
+	Method *MethodMatch `json:"method,omitempty"`
```

Comment:
lets make this plural with methods similar to headers to avoid duplicating user intent